internal/socket.go

```diff
@@ -70,9 +72,48 @@ func getUnixCreds(conn net.Conn) (*unix.Ucred, error) {
 		return nil, fmt.Errorf("doing rawConn Control: %w", err)
 	}
 
+	if err := authenticateUser(ucred); err != nil {
+		return nil, err
+	}
+
 	return ucred, nil
 }
 
+func authenticateUser(ucred *unix.Ucred) error {
+	// root?
+	if ucred.Uid == 0 {
+		return nil
+	}
+	userInfo, err := user.LookupId(fmt.Sprintf("%d", ucred.Uid))
+	if err != nil {
+		return fmt.Errorf("authenticate user, lookup user info: %s", err)
+	}
+	// user has sudo?
+	// #nosec G204 -- input is properly validated
+	out, err := exec.Command("sudo", "-l", "-U", userInfo.Username).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("authenticate user, check user has sudo: %s", err)
+	}
+	if !strings.Contains(string(out), "not allowed to run sudo") {
```

Comment:
what would happen if the OS is in another language than English?

Reply:
Good question. But there is no way to do simple `sudo` check. Maybe need to implement some combined workaround check.

Reply:
Left only check by `sudo` group.